bgpd/bgp_mplsvpn.c

```diff
@@ -608,7 +644,7 @@ void ensure_vrf_tovpn_sid(struct bgp *bgp_vpn, struct bgp *bgp_vrf, afi_t afi)
 	}
 
 	tovpn_sid_locator =
-		XCALLOC(MTYPE_BGP_SRV6_SID, sizeof(struct in6_addr));
+		XCALLOC(MTYPE_BGP_SRV6_SID, sizeof(struct srv6_locator_chunk));
```

Comment:
Should be MTYPE_SRV6_LOCATOR_CHUNK changed as well?

Reply:
I changed to use `srv6_locator_chunk_alloc` instead of `XCALLOC`. I also changed to use `srv6_locator_chunk_free` instead of `XFREE`.